### ISSUE TITLE
HiPathia

### ISSUE THREAD

**Issue by martahidalgo** [Bioconductor/Contributions#613]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/babelomics/hipathia

Confirm the following by editing each check box to '[x]'

- [ x ] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [ x ] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [ x ] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [ x ] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [ x ] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [ x ] The 'devel' branch for new packages and features.
- [ x ] The stable 'release' branch, made available every six
      months, for bug fixes.
- [ x ] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For help with submitting your package, please subscribe and post questions
to the [bioc-devel][4] mailing list.

[1]: https://bioconductor.org/developers/package-guidelines/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/

**Comment by martahidalgo** [comment]
AdditionalPackage: https://github.com/babelomics/hpAnnot

**Comment by martahidalgo** [comment]
Hi, 
Hipathia has an additional package on which it depends, hpAnnot. I have submitted it as a contribution too, #614. Sorry for the inconveniences.

**Comment by mtmorgan** [comment]
Does hpAnnot have the web hook established? Please bump it's version (to 0.99.1)

**Comment by martahidalgo** [comment]
Yes, hpAnnot should have the webhook. I have upgraded version to 0.99.1

**Comment by mtmorgan** [comment]
It should have triggered a 'build', can you look under 'Settings' 'Web hooks', find the section 'Recent Delivieries' and look at the 'response' for the most recent delivery?

**Comment by martahidalgo** [comment]
Hi,
I've updated both packages, hipathia (now version 0.99.2) and hpAnnot (on which hipathia depends, now version 0.99.3). hpAnnot must be installed before installing hipathia.
Thank you

**Comment by martahidalgo** [comment]
Hi, 
Sorry for the inconveniences, but what can I do to rebuild hpAnnot?
I've seen that when I push a new version of Hipathia, it triggers a build. However, when I push a new version of hpAnnot (on which hipathia depends) nothing happens. Since there was a problem in hpAnnot, which I hopefully solved, hipathia could not be installed. So in order to avoid the "skipped, ERROR" label from bioc-issue-bot, hpAnnot must be available before trying to build hipathia.
Thank you

**Comment by vobencha** [comment]
Hi,
I'll take a look at both packages today. Don't worry about the builds right now - we'll sort that out once I review them.
Valerie

**Comment by martahidalgo** [comment]
Ok, thank you!

**Comment by vobencha** [comment]
hiPatha is too large to be a data experiment package. I believe we're seeing the 'timeout' in the build because it takes longer than 10 minutes to download. I'm still downloading it so I can't yet comment on what you've got in there.

Options are to reduce the amount of data, store it more efficiently or submit hiPatha as an AnnotationHub or ExperimentHub package. See this page for details:

https://bioconductor.org/developers/package-guidelines/#package-types

**Comment by martahidalgo** [comment]
Hi, 
Hipathia is only 5.4Mb and the zip file to be downloaded from git is only 1.7Mb, so the problem must be in hpAnnot. 
Effectively, hpAnnot is about 1Gb, the zip file to be downloaded is 440Mb and takes a while to download. But I think the main problem is the "private" folder, with 730Mb. The rest of the package totals 121Mb. 
I'm going to reduce or delete the private folder, and then push a new version.

Thank you.

**Comment by vobencha** [comment]
Yes, the problem is with the data experiment package, hpAnnot. Sorry for the wrong name confusion.

**Comment by martahidalgo** [comment]
Hi, 
I've reduced the hpAnnot package and now for me it only takes 7 minutes to download from github. 
I hope it will be enough. 

Now hipathia gives an error in the vignette, but I think the problem is it does not find hpAnnot files, because it fails the first time a hpAnnot file needs to be load.

Thank you.

**Comment by vobencha** [comment]
Hi,

I've taken a look at both packages. Please see comments below.

### Summary

The data package should be put in ExperimentHub. Please work with lori.shepherd@roswellpark.org to do this. The vignette for the data package should explain how the data were downloaded, QCd and normalized.

If possible, the software package should only include a small number of annotations in inst/extdata, the rest should be downloaded by the user. The vignette for the software package should refer to the data package vignette for preparing the data; the rest should be the analysis as you have it.

### Software Package

1) The name of the package is hipatha but it's refered to as Hipatha throughout the software and data experiment package. Please change all references to hipatha.

2) Remove the private/ directory. This does not belong in an finalized R package.

3) Function names
The dot '.' is used for S3 dispatch. Please rename your functions with underscores or camel case.

4) Move the NEWS file to the same level as DESCRIPTION.

5) Please use BiocStyle to create the vignette:
https://bioconductor.org/packages/release/bioc/html/BiocStyle.html

6) Explain why these files are needed:
```
~/hipathia >ls inst/extdata/
pathway-viewer  report-files
```
7) Once the data structures are changed to SummarizedExperiment, MultiAssayExperiment, etc. object, the methods in hipatha need to be modified to take this object as input.

### Data Package

Unfortuately, 7 minutes to download a single package is not ok. The Bioconductor build system builds >1500 packages a night. Each package is allotted time dependent on the number of packages and computational resources available. If the package does not pass R CMD build or R CMD check within this time it will be flagged as a 'timeout'. It would be unfortunate for the software package to get a timeout simply due to the downloading of the data package.

1) design
It's not clear to me why so much static data are needed to demonstrate the use of the hipatha software package. In theory, hipatha "should work" on any properly normalized data, correct? It is not a given that the user will be working with the same genes you've saved in the data package.

    The current data package is a mixture of experimental data (hpAnnot/data/) and annotations (hpAnnot/inst/extdata/). These data types are usually separated in to annotation and experimental data packages; annotations often require updates as new data are released while experimental data are usually static.

    The gene expression data in hpAnnot/data should go in ExperimentHub. As for the annotations, I would ask if it's necessary to have all annotations in hpAnnot/inst/extdata/ in order for hipatha to function. Can you include one or two files for demonstration purposes and then instruct the user to download their own from a public source?

2) data storage class
Please reuse one of the existing classes designed to hold gene expression data. Options would be SummarizedExperiment or MultiAssayExperiment or a class from DESeq2 or edgeR. This will allow the data (brca_data) to be combined with the metadata (brca_design) and give you immediate access to getters and setters for data manipulation. Most of these classes hold the expression data in a matrix so it's just a matter of extracting that slot before calling the methods in hipatha.

3) The 'comp' and 'results' data objects should be produced real time and not saved. Why were these saved as static objects? Do they take very long to compute?

4) Please remove the private/ directory.

**Comment by martahidalgo** [comment]
Hi, 

Thank you for your feedback. I will address your suggestions. To answer your questions (and put some others):

Software Package:
1.- Can I change the package name to Hipathia? Or capital letters are not allowed?

6.- The files in the two folders in inst/extdata are a template of the HTML used to show dynamically the results of the analysis in a web browser. They are copied to the results folder when creating a results report and are needed to visualize the results.

Data Package:
1.- The additional Package hpAnnot was intended in fact as an Annotation package, hence its name. The data files were stored there only to make Hipathia lighter, there is no need to be in hpAnnot, they can be in Hipathia or in a new packae (or ExperimentHub). 
The files in inst/data are the annotations files for Hipathia and all of them are needed for Hipathia to work. These files are preprocessed data storing, among others, the pathways topologies and structures and the specific annotations between pathways and functions, for each of the supported species. If there is a better way to pack this data I have no problem to change it.

3.- "comp" and "results" data objects are saved in order to make quicker the checking of the examples. There are many functions which need these two kind of objects to work. Their computation is not very time-consuming once, but since they have to be computed once for (almost) each of the functions in the package, they increase the checking examples time dramatically. Maybe the examples can be written so that they are not executed (\dontrun).

Thank you

**Comment by vobencha** [comment]
Software
1) Yes caps are allowed and you can change the name of the package. If you do, we should close this issue and open a new one that points to the newly named github repo.
6) OK

Data Package
1) Since the big data are annotations, let's make hpAnnot an AnnotationHub package (not ExperimentHub). Please work on this package first. Once that package is approved you can open a new issue for Hipatha.
3) We don't encourage the use of /dontrun in examples so that's not a good alternative. Do you have an idea of how long the packages takes to pass R CMD check without generating these objects? Let's revisit this once the AnnotationHub package is done and we are reviewing the new version of Hipatha.

**Comment by vobencha** [comment]
@martahidalgo ,
I'm closing this issue. Once you have your data in S3, please open a new issue for the review of the hpAnnot package. Once hpAnnot is approved, open a new issue for Hipatha.

Valerie
